### DISCUSSION TITLE
having trouble with Hyper-v windows client

### DISCUSSION THREAD

**Discussion by Yonghong-Chen** [dorssel/usbipd-win#456] [Q&A] [answered]
Well, i am installing a windows guest on a windows host.

I can't change bios. So good old usbip need to turn on the testsigning is not going to work for me.

Then i am trying to use usbipd-win as server. Seems to work. Then usbipd-win does not come with a client, i guess i still need to use good old usbip as the client.

the client can "list" shared device id. but it wont attach to it.

It keeps giving me the error "already used bus id".

Also it seems strange to me, why the same device is reported as 
"8-2" by usbipd-win, but
"1-133" by usbip

Most thread were talking about a linux guest, there isn't that much information regarding windows guest. And I can't change the BIOS.  

Any help please?

**Comment by dorssel** [comment, marked as answer]
Like you stated: `usbipd-win` is only a server (USB device host). You will need other software on the client side. For Windows, there are several commercial and open source USBIP clients. As far as I know, none of the open source ones use signed drivers. If you need signed drivers, try one of the commercial solutions.

**Reply by Yonghong-Chen** [reply]
trying to use usbip as the client; 
And i ran usbipd on command console. when trying to attach from client i end up with 

      An exception occurred while communicating with the client:
      Usbipd.ConfigurationManagerException (1003): CM_Query_And_Remove_SubTree returned CR_REMOVE_VETOED: PNP_VetoOutstandingOpen, USB\VID_046D&PID_C21D\B3C8C2A1
         at Usbipd.ConfigurationManager.RestartingDevice..ctor(UInt32 ) in D:\a\usbipd-win\usbipd-win\Usbipd\ConfigurationManager.cs:line 383
         at Usbipd.ConfigurationManager.RestartingDevice..ctor(String ) in D:\a\usbipd-win\usbipd-win\Usbipd\ConfigurationManager.cs:line 366
         at Usbipd.ConnectedClient.HandleRequestImportAsync(CancellationToken ) in D:\a\usbipd-win\usbipd-win\Usbipd\ConnectedClient.cs:line 138

Seems unable to detach from host side.

Then i tried to bind it with "force", Same output.  
BTW, i am just trying to test this with an old model of LogicTech gamepad. 

C:\WINDOWS\system32>usbipd list
Connected:
BUSID  VID:PID    DEVICE                                                        STATE
2-10   27c6:63ac  Goodix Moc Fingerprint                                        Not shared
2-11   0c45:672e  Integrated Webcam                                             Not shared
2-14   8087:0026  Intel(R) Wireless Bluetooth(R)                                Not shared
5-4    0bda:8153  Realtek USB GbE Family Controller                             Not shared
7-5    413c:b06e  USB Input Device                                              Not shared
8-2    046d:c21d  Xbox 360 Controller for Windows                               Shared (forced)
9-1    047f:0127  USB Input Device                                              Not shared
9-3    046d:c534  USB Input Device                                              Not shared
9-5    413c:b06f  USB Input Device                                              Not shared

Persisted:
GUID                                  DEVICE
1f7d20ae-bd38-4687-b61f-8e0a6f8c9080  Xbox 360 Controller for Windows
Any light?

**Reply by dorssel** [reply]
> CR_REMOVE_VETOED: PNP_VetoOutstandingOpen

This means some software or driver is holding an exclusive lock to the device on Windows. Either stop that software first, or indeed use `usbipd bind --force`. That will permanently replace the driver (until you manually `usbipd unbind`), so Windows will not be able to use it. You may have to reboot as well if the device is still locked after the `--force`. After the reboot, the driver should be replaced, and Windows will not lock the device anymore.

> 8-2 046d:c21d Xbox 360 Controller for Windows Shared (forced)

This means that the `--force` was succesful. You should be able to attach that device from the client now.

> Persisted:
> GUID DEVICE
> 1f7d20ae-bd38-4687-b61f-8e0a6f8c9080 Xbox 360 Controller for Windows

This is probably the same device, but it was bound when it was plugged into a different port.

**Comment by mika76** [comment]
@dorssel Any chance of this becoming a client too in the future? It's enormously beneficial...

**Reply by dorssel** [reply]
No, see https://github.com/dorssel/usbipd-win/issues/65
